### PR TITLE
refact: add muxer for ivf, this is a start point for other muxer like mp4, webm

### PR DESCRIPTION
**Pull request by xuguangxin**

### CODE REVIEW

File: src/bin/rav1e.rs [hunk 1]
```diff
@@ -196,11 +192,11 @@ fn main() {
 
   if video_info.bit_depth == 8 {
     do_encode::<u8, y4m::Decoder<'_, Box<dyn Read>>>(
-      cfg, cli.verbose, progress, err, &mut cli.io.output, &mut source, y4m_enc
+      cfg, cli.verbose, progress, err, &mut *cli.io.output, &mut source, y4m_enc
```
Comment (lu-zero) [inline src/bin/rav1e.rs]: This probably will need a manual rebase.

Comment (lu-zero) [reply src/bin/rav1e.rs]: Let see if I managed to do it directly from the github UI

Comment (lu-zero) [reply src/bin/rav1e.rs]: No, it failed, I manually redid properly

Comment (xuguangxin) [reply src/bin/rav1e.rs]: Thanks for help.
Updated.

**Pull request merged**